STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/_az_ulib_dm_blob_ustream_interface.c

```diff
@@ -0,0 +1,358 @@
+
+  return AZ_OK;
+}
+
+static az_result concrete_get_position(az_ulib_ustream* ustream_instance, offset_t* const position)
+{
+  _az_PRECONDITION(AZ_ULIB_USTREAM_IS_TYPE_OF(ustream_instance, api));
+  _az_PRECONDITION_NOT_NULL(position);
+
+  return AZ_ERROR_ULIB_DISABLED;
+}
+
+static az_result concrete_release(az_ulib_ustream* ustream_instance, offset_t position)
+{
+  _az_PRECONDITION(AZ_ULIB_USTREAM_IS_TYPE_OF(ustream_instance, api));
+
+  return AZ_ERROR_ULIB_DISABLED;
+}
+
+static az_result concrete_clone(
+    az_ulib_ustream* ustream_instance_clone,
+    az_ulib_ustream* ustream_instance,
+    offset_t offset)
+{
+  _az_PRECONDITION(AZ_ULIB_USTREAM_IS_TYPE_OF(ustream_instance, api));
+  _az_PRECONDITION_NOT_NULL(ustream_instance_clone);
+
+  az_result result;
+
+  if (offset > (UINT32_MAX - ustream_instance->length))
+  {
+    result = AZ_ERROR_ARG;
+  }
+  else
+  {
+    init_instance(
+        ustream_instance_clone,
+        ustream_instance->control_block,
+        ustream_instance->inner_current_position,
+        offset,
+        ustream_instance->length);
+    result = AZ_OK;
+  }
+
+  return result;
+}
+
+static az_result concrete_dispose(az_ulib_ustream* ustream_instance)
+{
+  _az_PRECONDITION(AZ_ULIB_USTREAM_IS_TYPE_OF(ustream_instance, api));
+
+  az_ulib_ustream_data_cb* control_block = ustream_instance->control_block;
+
+  AZ_ULIB_PORT_ATOMIC_DEC_W(&(control_block->ref_count));
+  if (control_block->ref_count == 0)
+  {
+    destroy_control_block(control_block);
+  }
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_ulib_ustream_init(
+    az_ulib_ustream* ustream_instance,
+    az_ulib_ustream_data_cb* ustream_control_block,
+    az_ulib_release_callback control_block_release,
+    const uint8_t* const data_buffer,
+    size_t data_buffer_length,
+    az_ulib_release_callback data_buffer_release)
+{
+  _az_PRECONDITION_NOT_NULL(ustream_instance);
+  _az_PRECONDITION_NOT_NULL(ustream_control_block);
+  _az_PRECONDITION_NOT_NULL(data_buffer);
+  _az_PRECONDITION(data_buffer_length > 0);
+
+  ustream_control_block->api = &api;
+  ustream_control_block->ptr = (const az_ulib_ustream_data*)data_buffer;
+  ustream_control_block->ref_count = 0;
+  ustream_control_block->data_release = data_buffer_release;
+  ustream_control_block->control_block_release = control_block_release;
+
+  init_instance(ustream_instance, ustream_control_block, 0, 0, data_buffer_length);
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result create_ustream_from_blob(
+    az_ulib_ustream* ustream_instance_ptr,
+    az_ulib_ustream_data_cb* data_cb,
+    az_blob_http_cb* blob_http_cb,
+    NXD_ADDRESS* ip,
+    CHAR* resource,
+    CHAR* host,
+    az_ulib_release_callback control_block_release,
+    az_ulib_release_callback data_buffer_release)
+{
+  AZ_ULIB_TRY
+  {
+    // initialize blob http control block
+    blob_http_cb->_internal.ip = ip;
+    blob_http_cb->_internal.resource = resource;
+    blob_http_cb->_internal.host = host;
```

Comment:
Consider remove `ip`, `resource`, and `host` from the `blob_http_cb`, only provide it directly to `az_nx_blob_client_init`. You are not using it anywhere else.